### PR TITLE
Include actual value for date compare assertions

### LINKED ISSUE #7031: Display source time in time comparaison exceptions
In [NotBeforeTimeException](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/javatool/base/main/NotBeforeTimeException.java.mustache) and [NotAfterTimeException](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/javatool/base/main/NotAfterTimeException.java.mustache) we are only displaying one of the two compared times which can result in harder analysis when running in production.

We should display both times (that mean reworking the building logic of those exceptions and updating [Assert](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/javatool/base/main/Assert.java.mustache))

### PR DESCRIPTION
**Pull request by adem-hassine**
Close #7031

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java [hunk 1]
```diff
@@ -8,19 +8,25 @@ private NotAfterTimeException(String field, String message) {
     super(field, message);
   }
 
-  public static NotAfterTimeException notStrictlyAfter(String fieldName, Instant other) {
-    return new NotAfterTimeException(fieldName, message(fieldName, "must be strictly after", other));
+  public static NotAfterTimeException notStrictlyAfter(String fieldName, Instant actual, Instant other) {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java:11]: I don't like having 2 parameters of the same in a public method. Those should be refactored to something without ambiguity (I'd go for a fluent builder as all parameters here are mandatory)

**Review by DamnClin — CHANGES_REQUESTED** [review]

File: src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java (hunk 1 above)
Comment (adem-hassine) [reply src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java:11]: So, do you want to introduce the builder pattern for all the fields or only the new one?

Comment (DamnClin) [reply src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java:11]: Must be for all fields but it's not just a builder, it's one where you can't miss any field at compile time

Comment (adem-hassine) [reply src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java]: Please check again and let me know if this is good. If it is, I can add some blank or empty assertions in the final build function.

File: src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java [hunk 2]
```diff
@@ -8,24 +8,40 @@ private NotAfterTimeException(String field, String message) {
...
-    return new NotAfterTimeException(fieldName, message(fieldName, "must be strictly after", other));
+  public static NotAfterTimeExceptionBuilder field(String fieldName, Instant value) {
+    return new NotAfterTimeExceptionBuilder(fieldName, value);
   }
 
-  public static NotAfterTimeException notAfter(String fieldName, Instant other) {
-    return new NotAfterTimeException(fieldName, message(fieldName, "must be after", other));
-  }
+  record NotAfterTimeExceptionBuilder(String fieldName, Instant value) {
+    public NotAfterTimeException notStrictlyAfter(Instant other) {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java:16]: strictlyNotAfter sounds better

File: src/main/java/tech/jhipster/lite/error/domain/NotBeforeTimeException.java [hunk 3]
```diff
@@ -8,24 +8,40 @@ private NotBeforeTimeException(String field, String message) {
...
-    return new NotBeforeTimeException(fieldName, message(fieldName, "must be strictly before", other));
+  public static NotBeforeTimeException.NotBeforeTimeExceptionBuilder field(String fieldName, Instant value) {
+    return new NotBeforeTimeException.NotBeforeTimeExceptionBuilder(fieldName, value);
   }
 
-  public static NotBeforeTimeException notBefore(String fieldName, Instant other) {
-    return new NotBeforeTimeException(fieldName, message(fieldName, "must be before", other));
-  }
+  public record NotBeforeTimeExceptionBuilder(String fieldName, Instant value) {
+    public NotBeforeTimeException notStrictlyBefore(Instant other) {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/error/domain/NotBeforeTimeException.java:16]: strictlyNotBefore sounds better

**Review by DamnClin — COMMENTED** [review]
I like it but formatting is off, you should build an app with java-base and prettier, then run prettier (`npm i` and `npm run prettier:format`) to get a formatted version of the templates

**Comment by DamnClin** [comment]
You should have either Fix #issue or Close #issue in your PR comment so the issue will close when merging it

File: src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java [hunk 4]
```diff
@@ -8,24 +8,40 @@ private NotAfterTimeException(String field, String message) {
...
-      .append(" ")
-      .append(other)
-      .append(" but wasn't")
-      .toString();
+    private static String message(String fieldName, Instant actual, String hint, Instant other) {
+      return new StringBuilder()
+        .append("Time in \"")
+        .append(fieldName)
+        .append("\" ")
+        .append("having value")
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java:33]: can update to "having value : " and remove the next lines :)

File: src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java (hunk 1 above)
Comment (DamnClin) [reply src/main/java/tech/jhipster/lite/error/domain/NotAfterTimeException.java]: I think they must be in the record constructor

**Review by DamnClin — APPROVED** [review]

**Pull request merged**